lib/index.d.ts

```diff
@@ -817,6 +828,14 @@ interface NamedSubscription {
     timeToLive: number;
 }
 
+type ExtractPropertyNamesOfType<T, S> = {
+    [K in keyof T]: T[K] extends S ? K : never
+}[keyof T];
```

Comment:
I for one (obviously 🤔 ) know what it does - but for future maintainers of the code, we might need a small explanation of this piece of black magic.

Reply:
Comments added